applications/storage_settings/scenes/storage_settings_scene_sd_info.c

```diff
@@ -24,13 +24,44 @@ void storage_settings_scene_sd_info_on_enter(void* context) {
             dialog_ex, "Try to reinsert\nor format SD\ncard.", 3, 19, AlignLeft, AlignTop);
         dialog_ex_set_center_button_text(dialog_ex, "Ok");
     } else {
+        char unit_kb[] = "KB";
+        char unit_mb[] = "MB";
+        char unit_gb[] = "GB";
+
+        double sd_total_val = (double)sd_info.kb_total;
```

Comment:
Using float point math here is kinda overkill.

Reply:
Especially using double. Float should be more than enough, use cast at the last moment when passing it to string_printf.